src/docker.c

```diff
@@ -36,10 +36,11 @@ void init_curl(DOCKER *client) {
   curl_easy_setopt(client->curl, CURLOPT_WRITEDATA, client->buffer);
 }
 
-CURLcode perform(DOCKER *client, char *url) {
+CURLcode perform(DOCKER *client, char *url, long *http_status) {
   init_buffer(client);
   curl_easy_setopt(client->curl, CURLOPT_URL, url);
   CURLcode response = curl_easy_perform(client->curl);
+  curl_easy_getinfo(client->curl, CURLINFO_RESPONSE_CODE, http_status);
```

Comment:
What's your reasoning for using a required pointer param and always fetching the response code, vs using an optional pointer param and conditionally fetching the response code? Doing it this way will have fewer branches but potentially do more work?

curl_easy_getinfo looks like it is somewhat nontrivial, that is, it's not just a pointer offset into a data structure using the CURLINFO value or something like that; it performs two switches in order to find the correct field. First it switches on a type code that's part of the CURLINFO, then it switches on the actual CURLINFO within the type-specific part to pull the correct value https://github.com/curl/curl/blob/dafdb20a26d0c890e83dea61a104b75408481ebd/lib/getinfo.c#L562

So in the case of not wanting the status code it might be faster to skip the call to getinfo, rather than retrieve it and discard it e.g. `if (http_status) { curl_easy_getinfo(...`

Reply:
Included this in the same commit as above.